packages/manager/apps/pci-private-network/src/data/hooks/networks/useNetworks.ts

```diff
@@ -53,3 +150,48 @@ export const useSubnets = (
     queryFn: () => getSubnets(projectId, region, networkId),
     enabled: !!(projectId && networkId && region),
   });
+
+export const updatePrivateNetworksList = async (
+  projectId: string,
+  newNetwork: TNetwork,
+) => {
+  const queryKey = networksQueryKey(projectId);
+  const networks = queryClient.getQueryData<TNetwork[]>(queryKey);
+
+  if (networks) {
+    queryClient.setQueryData(queryKey, [newNetwork, ...networks]);
+  }
+};
+
+type DeletePrivateNetworkProps = {
+  projectId: string;
+  region: string;
+  networkId: string;
+  onError: (error: Error) => void;
+  onSuccess: () => void;
+};
+
+export const useDeletePrivateNetwork = ({
+  projectId,
+  region,
+  networkId,
+  onError,
+  onSuccess,
+}: DeletePrivateNetworkProps) => {
+  const mutation = useMutation({
+    mutationFn: () => apiDeleteNetwork(projectId, region, networkId),
+    onError,
+    onSuccess: async () => {
+      const queryKey = networksQueryKey(projectId);
+      queryClient.setQueryData(queryKey, (networks: TNetwork[]) =>
+        networks.filter((network) => network.id !== networkId),
+      );
+      onSuccess();
+    },
+  });
+
+  return {
+    deletePrivateNetwork: () => mutation.mutate(),
```

Comment:
```suggestion
    deletePrivateNetwork: mutation.mutate,
```
I think this would avoid creating a new function each time, it doesn't seem to be used in other projects, maybe there is a reason

Reply:
Also do you really need to encapsulate the useMutation result in a new object ?
Using it like this would result in the same no ?
`
const { mutate: deletePrivateNetwork } = useDeletePrivateNetwork(param);
`

Reply:
You right